OpenDreamShared/Compiler/DM/DMParser.cs

```diff
@@ -919,6 +923,7 @@ public DMASTProcStatement ProcStatement()
                 DMASTExpression conditional = Expression();
                 if (conditional == null) Error("Expected conditional");
                 ConsumeRightParenthesis();
+                Check(TokenType.DM_Semicolon);
```

Comment:
This doesn't seem necessary for a do/while

Reply:
```suggestion
```